lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -280,6 +282,20 @@ def from_requirement_dict(cls, collection_req, art_mgr):
                 req_type = 'file'
                 req_source = src_path
             elif _is_collection_dir(src_path):
+                if _is_installed_collection_dir(src_path) and _is_collection_src_dir(src_path):
+                    # Note that ``download`` requires a dir with a ``galaxy.yml`` and fails if it
+                    # doesn't exist, but if a ``MANIFEST.json`` also exists, it would be used
+                    # instead of the ``galaxy.yml``.
+                    raise AnsibleError(
+                        u"Collection requirement at '{path!s}' has both a {manifest_json!s} "
+                        u"file and a {galaxy_yml!s}.\nThe requirement must either be an artifact "
```

Comment:
I think we usually use "artifact" to talk about publishable tarballs:
```suggestion
                        u"file and a {galaxy_yml!s}.\nThe requirement must either be an installed "
```